polar-core/src/kb.rs

```diff
@@ -551,6 +551,28 @@ impl KnowledgeBase {
         Ok(())
     }
 
+    /// Return true if a constant with the given name has been defined.
+    pub fn is_constant(&self, name: &Symbol) -> bool {
+        self.constants.contains_key(name)
+    }
+
+    /// Getter for `constants` map without exposing it for mutation.
+    pub fn get_registered_constants(&self) -> &Bindings {
+        &self.constants
+    }
+
+    pub fn get_registered_constant(&self, constant: &Term) -> PolarResult<&Term> {
```

Comment:
Having APIs that only accept one term type but don't say so in the type description is unfortunate. You are probably already thinking this from your comments, but we should rethink how we handle source information & types so that it can be attached to more specific types so that our APIs can be more clear.

Could you write a clean-up task about this (assuming we're on the same page)?

Reply:
https://www.notion.so/osohq/Allow-Term-to-be-specific-over-the-inner-Value-type-0c6fa761e84b4b03a1a5f1be83c1d137

Reply:
o